src/core/ddsi/src/ddsi_config.c

```diff
@@ -680,9 +688,7 @@ static int if_psmx(struct ddsi_cfgst *cfgst, void *parent, struct cfgelem const
   struct ddsi_config_psmx_listelem *new = if_common (cfgst, parent, cfgelem, sizeof(*new));
   if (new == NULL)
     return -1;
-  new->cfg.name = NULL;
```

Comment:
`if_common` at some point started doing a `memset` — it is quite amazing it avoided doing it for as many years as it did! — so rather than adding a `memset` here, we should just delete setting the fields to null pointers.

See https://github.com/eclipse-cyclonedds/cyclonedds/blob/00c63c342c2d858bf94b6a602dcf8bd696fdbe26/src/core/ddsi/src/ddsi_config.c#L649, or commit e6fc9a9ac5f389cded25057ae909e90ad2b5c152.

Reply:
I added a commit removing some of the redundant initialisations, at least all the ones I could find with a quick search.